src/openms_gui/source/VISUAL/APPLICATIONS/TOPPViewBase.cpp

```diff
@@ -1872,6 +1876,16 @@ namespace OpenMS
     return &(canvas->getCurrentLayer());
   }
 
+  LayerData* TOPPViewBase::getCurrentLayer()
```

Comment:
I think this TOPPViewBase should not have a function to get the "currentLayer".
The way it is done with getActiveCanvas()->getCurrentLayer() also not ideal.
I think cleaner would be to get the activeWorkspace and try to call the annotation there

Reply:
Agreed, this is not ideal. 
This was simply factoring out of some code, which should make it easier to replace in the future.
But one step at a time.